### PR TITLE
Fold AIEVec shift and broadcast ops to a new AIEVec broadcast op

### PR DESCRIPTION
**Pull request by linay-xsj**

### CODE REVIEW

**Comment by linay-xsj** [comment]
This changeset converts 
%1 = aievec.shift %0 {shift = 16 : i32} : vector<32xi8>, vector<32xi8>
%2 = aievec.broadcast %1 {idx = 0 : i8} : vector<32xi8>, vector<32xi8>
to 
%1 = aievec.broadcast %0 {idx = 16 : i8} : vector<32xi8>, vector<32xi8>

**Comment by linay-xsj** [comment]
This changeset modified the heuristic of addressing the unaligned transfer read with a constant innermost index. We
get the corresponding aligned load and its next vector load followed by an aievec.shift op.
Example:
Convert -
%0 = vector.transfer_read %arg1[16] : vector<32xi8>
%1 = vector.transfer_read %arg1[34] : vector<32xi8>

to -

%0 = aievec.upd %arg1[0] : vector<32xi8>
%1 = aievec.upd %arg1[32] : vector<32xi8>
%2 = aievec.shift %0, %1 {shift = 16 : i32} : vector<32xi8>
%3 = aievec.upd %arg1[64] : vector<32xi8>
%4 = aievec.shift %1, %3 {shift = 2 : i32} : vector<32xi8>

**Review by david-vc — APPROVED** [review]
Last time we discussed the handling of unaligned transfer_read with a constant address we thought it would be better to handle this purely within the vector dialect. I think that this could be accomplished with vector.extract_srided slice and vector.insert_strided_slice. Instead you are dealing as part of LowerVectorTransferReadToAIEUPD. 
The code changes look good.

File: lib/Dialect/AIEVec/Transforms/ConvertVectorToAIEVec.cpp [hunk 1]
```diff
@@ -588,6 +641,34 @@ static void configureAIEVecV2Legalizations(ConversionTarget &target,
...
 }
 
+static void
+configureAIEVecV1TransformationLegalizations(ConversionTarget &target) {}
+
+static void
+configureAIEVecV2TransformationLegalizations(ConversionTarget &target) {
+  target.addDynamicallyLegalOp<xilinx::aievec::BroadcastOp>(
+      [](xilinx::aievec::BroadcastOp op) {
+        if (!op.getSource().getDefiningOp())
```
Comment (david-vc) [inline lib/Dialect/AIEVec/Transforms/ConvertVectorToAIEVec.cpp:651]: The code here (651-666) duplicates the code in FoldAIEShiftAndBroadcast::matchAndRewrite.
Is this expected?
Should we factor this out? The main difference is just the return type, but that does not prevent refactoring this.

**Pull request merged**